File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/SearchClusterLogsDetails.java
package com.oracle.aidataplatform.dp.model;



/**
 * Request payload for searching logs in a cluster.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=SearchClusterLogsDetails.Builder.class)

public final class SearchClusterLogsDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"timeBegin", "timeEnd", "logContentTypeContains", "subjectContains", "messageContains", "logLevel", "opcRequestId", "advancedFilter", "thread", "executionContextId", "logStreamTypeContains", "eventType"})
    public SearchClusterLogsDetails(java.util.Date timeBegin, java.util.Date timeEnd, String logContentTypeContains, String subjectContains, String messageContains, String logLevel, String opcRequestId, String advancedFilter, String thread, String executionContextId, String logStreamTypeContains, String eventType) {
        super();
        this.timeBegin = timeBegin;
        this.timeEnd = timeEnd;
        this.logContentTypeContains = logContentTypeContains;
        this.subjectContains = subjectContains;
        this.messageContains = messageContains;
        this.logLevel = logLevel;
        this.opcRequestId = opcRequestId;
        this.advancedFilter = advancedFilter;
        this.thread = thread;
        this.executionContextId = executionContextId;
        this.logStreamTypeContains = logStreamTypeContains;
        this.eventType = eventType;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Start filter log's date and time, in RFC 3339 format.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeBegin")
private java.util.Date timeBegin;

        /**
         * Start filter log's date and time, in RFC 3339 format.
         * @param timeBegin the value to set
         * @return this builder
         **/
        

public Builder timeBegin(java.util.Date timeBegin) {
    this.timeBegin = timeBegin;
    return this;
}
            /**
     * End filter log's date and time, in RFC 3339 format.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeEnd")
private java.util.Date timeEnd;

        /**
         * End filter log's date and time, in RFC 3339 format.
         * @param timeEnd the value to set
         * @return this builder
         **/
        

public Builder timeEnd(java.util.Date timeEnd) {
    this.timeEnd = timeEnd;
    return this;
}
            /**
     * The string represent the type of compute logs 
* Common values include {@code events}, {@code driver}, {@code executor}.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("logContentTypeContains")
private String logContentTypeContains;

        /**
         * The string represent the type of compute logs 
* Common values include {@code events}, {@code driver}, {@code executor}.
* 
         * @param logContentTypeContains the value to set
         * @return this builder
         **/
        

public Builder logContentTypeContains(String logContentTypeContains) {
    this.logContentTypeContains = logContentTypeContains;
    return this;
}
            /**
     * The string represent the specific of compute nodes
* Common values include {@code spark-driver}, {@code spark-executor-1}.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("subjectContains")
private String subjectContains;

        /**
         * The string represent the specific of compute nodes
* Common values include {@code spark-driver}, {@code spark-executor-1}.
* 
         * @param subjectContains the value to set
         * @return this builder
         **/
        

public Builder subjectContains(String subjectContains) {
    this.subjectContains = subjectContains;
    return this;
}
            /**
     * The string represent a filter to search message string from the log
* Supported only when {@code logContentTypeContains} is {@code events}.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("messageContains")
private String messageContains;

        /**
         * The string represent a filter to search message string from the log
* Supported only when {@code logContentTypeContains} is {@code events}.
* 
         * @param messageContains the value to set
         * @return this builder
         **/
        

public Builder messageContains(String messageContains) {
    this.messageContains = messageContains;
    return this;
}
            /**
     * The severity level of the log messages to filter. 
* Common values include {@code INFO}, {@code WARN}, {@code ERROR}, {@code DEBUG}, etc.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("logLevel")
private String logLevel;

        /**
         * The severity level of the log messages to filter. 
* Common values include {@code INFO}, {@code WARN}, {@code ERROR}, {@code DEBUG}, etc.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
         * @param logLevel the value to set
         * @return this builder
         **/
        

public Builder logLevel(String logLevel) {
    this.logLevel = logLevel;
    return this;
}
            /**
     * The unique request identifier used for tracking the API call.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("opcRequestId")
private String opcRequestId;

        /**
         * The unique request identifier used for tracking the API call.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
         * @param opcRequestId the value to set
         * @return this builder
         **/
        

public Builder opcRequestId(String opcRequestId) {
    this.opcRequestId = opcRequestId;
    return this;
}
            /**
     * Contains any key and value that you want to search in the logs.
* Should be provided in the format 'key = value'. As of now supported keys are 'taskRunKey' and 'jobRunKey'
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("advancedFilter")
private String advancedFilter;

        /**
         * Contains any key and value that you want to search in the logs.
* Should be provided in the format 'key = value'. As of now supported keys are 'taskRunKey' and 'jobRunKey'
* 
         * @param advancedFilter the value to set
         * @return this builder
         **/
        

public Builder advancedFilter(String advancedFilter) {
    this.advancedFilter = advancedFilter;
    return this;
}
            /**
     * The thread identifier that generated the log.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("thread")
private String thread;

        /**
         * The thread identifier that generated the log.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
         * @param thread the value to set
         * @return this builder
         **/
        

public Builder thread(String thread) {
    this.thread = thread;
    return this;
}
            /**
     * The execution context identifier associated with the log.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("executionContextId")
private String executionContextId;

        /**
         * The execution context identifier associated with the log.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
         * @param executionContextId the value to set
         * @return this builder
         **/
        

public Builder executionContextId(String executionContextId) {
    this.executionContextId = executionContextId;
    return this;
}
            /**
     * Filters logs based on the output stream type. 
* - This includes standard output ({@code stdout}) for regular program output 
* - and standard error ({@code stderr}) for error messages and diagnostics.
* Only supported when logContentTypeContains=executor/driver.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("logStreamTypeContains")
private String logStreamTypeContains;

        /**
         * Filters logs based on the output stream type. 
* - This includes standard output ({@code stdout}) for regular program output 
* - and standard error ({@code stderr}) for error messages and diagnostics.
* Only supported when logContentTypeContains=executor/driver.
* 
         * @param logStreamTypeContains the value to set
         * @return this builder
         **/
        

public Builder logStreamTypeContains(String logStreamTypeContains) {
    this.logStreamTypeContains = logStreamTypeContains;
    return this;
}
            /**
     * Filters logs based on the event type.
* Supported only when {@code logContentTypeContains} is {@code events}.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("eventType")
private String eventType;

        /**
         * Filters logs based on the event type.
* Supported only when {@code logContentTypeContains} is {@code events}.
* 
         * @param eventType the value to set
         * @return this builder
         **/
        

public Builder eventType(String eventType) {
    this.eventType = eventType;
    return this;
}


        public SearchClusterLogsDetails build() {
            SearchClusterLogsDetails model = new SearchClusterLogsDetails(this.timeBegin
                , this.timeEnd
                , this.logContentTypeContains
                , this.subjectContains
                , this.messageContains
                , this.logLevel
                , this.opcRequestId
                , this.advancedFilter
                , this.thread
                , this.executionContextId
                , this.logStreamTypeContains
                , this.eventType);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(SearchClusterLogsDetails model) {
                this.timeBegin(model.getTimeBegin());
    this.timeEnd(model.getTimeEnd());
    this.logContentTypeContains(model.getLogContentTypeContains());
    this.subjectContains(model.getSubjectContains());
    this.messageContains(model.getMessageContains());
    this.logLevel(model.getLogLevel());
    this.opcRequestId(model.getOpcRequestId());
    this.advancedFilter(model.getAdvancedFilter());
    this.thread(model.getThread());
    this.executionContextId(model.getExecutionContextId());
    this.logStreamTypeContains(model.getLogStreamTypeContains());
    this.eventType(model.getEventType());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Start filter log's date and time, in RFC 3339 format.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeBegin")
    private final java.util.Date timeBegin;

        /**
     * Start filter log's date and time, in RFC 3339 format.
     * @return the value
     **/
    
    public java.util.Date getTimeBegin() {
        return timeBegin;
    }


        /**
     * End filter log's date and time, in RFC 3339 format.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeEnd")
    private final java.util.Date timeEnd;

        /**
     * End filter log's date and time, in RFC 3339 format.
     * @return the value
     **/
    
    public java.util.Date getTimeEnd() {
        return timeEnd;
    }


        /**
     * The string represent the type of compute logs 
* Common values include {@code events}, {@code driver}, {@code executor}.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("logContentTypeContains")
    private final String logContentTypeContains;

        /**
     * The string represent the type of compute logs 
* Common values include {@code events}, {@code driver}, {@code executor}.
* 
     * @return the value
     **/
    
    public String getLogContentTypeContains() {
        return logContentTypeContains;
    }


        /**
     * The string represent the specific of compute nodes
* Common values include {@code spark-driver}, {@code spark-executor-1}.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("subjectContains")
    private final String subjectContains;

        /**
     * The string represent the specific of compute nodes
* Common values include {@code spark-driver}, {@code spark-executor-1}.
* 
     * @return the value
     **/
    
    public String getSubjectContains() {
        return subjectContains;
    }


        /**
     * The string represent a filter to search message string from the log
* Supported only when {@code logContentTypeContains} is {@code events}.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("messageContains")
    private final String messageContains;

        /**
     * The string represent a filter to search message string from the log
* Supported only when {@code logContentTypeContains} is {@code events}.
* 
     * @return the value
     **/
    
    public String getMessageContains() {
        return messageContains;
    }


        /**
     * The severity level of the log messages to filter. 
* Common values include {@code INFO}, {@code WARN}, {@code ERROR}, {@code DEBUG}, etc.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("logLevel")
    private final String logLevel;

        /**
     * The severity level of the log messages to filter. 
* Common values include {@code INFO}, {@code WARN}, {@code ERROR}, {@code DEBUG}, etc.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     * @return the value
     **/
    
    public String getLogLevel() {
        return logLevel;
    }


        /**
     * The unique request identifier used for tracking the API call.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("opcRequestId")
    private final String opcRequestId;

        /**
     * The unique request identifier used for tracking the API call.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     * @return the value
     **/
    
    public String getOpcRequestId() {
        return opcRequestId;
    }


        /**
     * Contains any key and value that you want to search in the logs.
* Should be provided in the format 'key = value'. As of now supported keys are 'taskRunKey' and 'jobRunKey'
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("advancedFilter")
    private final String advancedFilter;

        /**
     * Contains any key and value that you want to search in the logs.
* Should be provided in the format 'key = value'. As of now supported keys are 'taskRunKey' and 'jobRunKey'
* 
     * @return the value
     **/
    
    public String getAdvancedFilter() {
        return advancedFilter;
    }


        /**
     * The thread identifier that generated the log.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("thread")
    private final String thread;

        /**
     * The thread identifier that generated the log.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     * @return the value
     **/
    
    public String getThread() {
        return thread;
    }


        /**
     * The execution context identifier associated with the log.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("executionContextId")
    private final String executionContextId;

        /**
     * The execution context identifier associated with the log.
* Supported only when {@code logContentTypeContains} is {@code executor} or {@code driver}.
* 
     * @return the value
     **/
    
    public String getExecutionContextId() {
        return executionContextId;
    }


        /**
     * Filters logs based on the output stream type. 
* - This includes standard output ({@code stdout}) for regular program output 
* - and standard error ({@code stderr}) for error messages and diagnostics.
* Only supported when logContentTypeContains=executor/driver.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("logStreamTypeContains")
    private final String logStreamTypeContains;

        /**
     * Filters logs based on the output stream type. 
* - This includes standard output ({@code stdout}) for regular program output 
* - and standard error ({@code stderr}) for error messages and diagnostics.
* Only supported when logContentTypeContains=executor/driver.
* 
     * @return the value
     **/
    
    public String getLogStreamTypeContains() {
        return logStreamTypeContains;
    }


        /**
     * Filters logs based on the event type.
* Supported only when {@code logContentTypeContains} is {@code events}.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("eventType")
    private final String eventType;

        /**
     * Filters logs based on the event type.
* Supported only when {@code logContentTypeContains} is {@code events}.
* 
     * @return the value
     **/
    
    public String getEventType() {
        return eventType;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("SearchClusterLogsDetails(");
        sb.append("timeBegin=").append(String.valueOf(this.timeBegin));
        sb.append(", timeEnd=").append(String.valueOf(this.timeEnd));
        sb.append(", logContentTypeContains=").append(String.valueOf(this.logContentTypeContains));
        sb.append(", subjectContains=").append(String.valueOf(this.subjectContains));
        sb.append(", messageContains=").append(String.valueOf(this.messageContains));
        sb.append(", logLevel=").append(String.valueOf(this.logLevel));
        sb.append(", opcRequestId=").append(String.valueOf(this.opcRequestId));
        sb.append(", advancedFilter=").append(String.valueOf(this.advancedFilter));
        sb.append(", thread=").append(String.valueOf(this.thread));
        sb.append(", executionContextId=").append(String.valueOf(this.executionContextId));
        sb.append(", logStreamTypeContains=").append(String.valueOf(this.logStreamTypeContains));
        sb.append(", eventType=").append(String.valueOf(this.eventType));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof SearchClusterLogsDetails)) {
            return false;
        }

        SearchClusterLogsDetails other = (SearchClusterLogsDetails) o;
        return java.util.Objects.equals(this.timeBegin, other.timeBegin) &&
            java.util.Objects.equals(this.timeEnd, other.timeEnd) &&
            java.util.Objects.equals(this.logContentTypeContains, other.logContentTypeContains) &&
            java.util.Objects.equals(this.subjectContains, other.subjectContains) &&
            java.util.Objects.equals(this.messageContains, other.messageContains) &&
            java.util.Objects.equals(this.logLevel, other.logLevel) &&
            java.util.Objects.equals(this.opcRequestId, other.opcRequestId) &&
            java.util.Objects.equals(this.advancedFilter, other.advancedFilter) &&
            java.util.Objects.equals(this.thread, other.thread) &&
            java.util.Objects.equals(this.executionContextId, other.executionContextId) &&
            java.util.Objects.equals(this.logStreamTypeContains, other.logStreamTypeContains) &&
            java.util.Objects.equals(this.eventType, other.eventType);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.timeBegin == null ? 43 : this.timeBegin.hashCode());
        result = (result * PRIME) + (this.timeEnd == null ? 43 : this.timeEnd.hashCode());
        result = (result * PRIME) + (this.logContentTypeContains == null ? 43 : this.logContentTypeContains.hashCode());
        result = (result * PRIME) + (this.subjectContains == null ? 43 : this.subjectContains.hashCode());
        result = (result * PRIME) + (this.messageContains == null ? 43 : this.messageContains.hashCode());
        result = (result * PRIME) + (this.logLevel == null ? 43 : this.logLevel.hashCode());
        result = (result * PRIME) + (this.opcRequestId == null ? 43 : this.opcRequestId.hashCode());
        result = (result * PRIME) + (this.advancedFilter == null ? 43 : this.advancedFilter.hashCode());
        result = (result * PRIME) + (this.thread == null ? 43 : this.thread.hashCode());
        result = (result * PRIME) + (this.executionContextId == null ? 43 : this.executionContextId.hashCode());
        result = (result * PRIME) + (this.logStreamTypeContains == null ? 43 : this.logStreamTypeContains.hashCode());
        result = (result * PRIME) + (this.eventType == null ? 43 : this.eventType.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/search_cluster_logs_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class SearchClusterLogsDetails(object):
    """
    Request payload for searching logs in a cluster.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new SearchClusterLogsDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param time_begin:
            The value to assign to the time_begin property of this SearchClusterLogsDetails.
        :type time_begin: datetime

        :param time_end:
            The value to assign to the time_end property of this SearchClusterLogsDetails.
        :type time_end: datetime

        :param log_content_type_contains:
            The value to assign to the log_content_type_contains property of this SearchClusterLogsDetails.
        :type log_content_type_contains: str

        :param subject_contains:
            The value to assign to the subject_contains property of this SearchClusterLogsDetails.
        :type subject_contains: str

        :param message_contains:
            The value to assign to the message_contains property of this SearchClusterLogsDetails.
        :type message_contains: str

        :param log_level:
            The value to assign to the log_level property of this SearchClusterLogsDetails.
        :type log_level: str

        :param opc_request_id:
            The value to assign to the opc_request_id property of this SearchClusterLogsDetails.
        :type opc_request_id: str

        :param advanced_filter:
            The value to assign to the advanced_filter property of this SearchClusterLogsDetails.
        :type advanced_filter: str

        :param thread:
            The value to assign to the thread property of this SearchClusterLogsDetails.
        :type thread: str

        :param execution_context_id:
            The value to assign to the execution_context_id property of this SearchClusterLogsDetails.
        :type execution_context_id: str

        :param log_stream_type_contains:
            The value to assign to the log_stream_type_contains property of this SearchClusterLogsDetails.
        :type log_stream_type_contains: str

        :param event_type:
            The value to assign to the event_type property of this SearchClusterLogsDetails.
        :type event_type: str

        """
        self.swagger_types = {
            'time_begin': 'datetime',
            'time_end': 'datetime',
            'log_content_type_contains': 'str',
            'subject_contains': 'str',
            'message_contains': 'str',
            'log_level': 'str',
            'opc_request_id': 'str',
            'advanced_filter': 'str',
            'thread': 'str',
            'execution_context_id': 'str',
            'log_stream_type_contains': 'str',
            'event_type': 'str'
        }

        self.attribute_map = {
            'time_begin': 'timeBegin',
            'time_end': 'timeEnd',
            'log_content_type_contains': 'logContentTypeContains',
            'subject_contains': 'subjectContains',
            'message_contains': 'messageContains',
            'log_level': 'logLevel',
            'opc_request_id': 'opcRequestId',
            'advanced_filter': 'advancedFilter',
            'thread': 'thread',
            'execution_context_id': 'executionContextId',
            'log_stream_type_contains': 'logStreamTypeContains',
            'event_type': 'eventType'
        }

        self._time_begin = None
        self._time_end = None
        self._log_content_type_contains = None
        self._subject_contains = None
        self._message_contains = None
        self._log_level = None
        self._opc_request_id = None
        self._advanced_filter = None
        self._thread = None
        self._execution_context_id = None
        self._log_stream_type_contains = None
        self._event_type = None

    @property
    def time_begin(self):
        """
        **[Required]** Gets the time_begin of this SearchClusterLogsDetails.
        Start filter log's date and time, in RFC 3339 format.


        :return: The time_begin of this SearchClusterLogsDetails.
        :rtype: datetime
        """
        return self._time_begin

    @time_begin.setter
    def time_begin(self, time_begin):
        """
        Sets the time_begin of this SearchClusterLogsDetails.
        Start filter log's date and time, in RFC 3339 format.


        :param time_begin: The time_begin of this SearchClusterLogsDetails.
        :type: datetime
        """
        self._time_begin = time_begin

    @property
    def time_end(self):
        """
        **[Required]** Gets the time_end of this SearchClusterLogsDetails.
        End filter log's date and time, in RFC 3339 format.


        :return: The time_end of this SearchClusterLogsDetails.
        :rtype: datetime
        """
        return self._time_end

    @time_end.setter
    def time_end(self, time_end):
        """
        Sets the time_end of this SearchClusterLogsDetails.
        End filter log's date and time, in RFC 3339 format.


        :param time_end: The time_end of this SearchClusterLogsDetails.
        :type: datetime
        """
        self._time_end = time_end

    @property
    def log_content_type_contains(self):
        """
        **[Required]** Gets the log_content_type_contains of this SearchClusterLogsDetails.
        The string represent the type of compute logs
        Common values include `events`, `driver`, `executor`.


        :return: The log_content_type_contains of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._log_content_type_contains

    @log_content_type_contains.setter
    def log_content_type_contains(self, log_content_type_contains):
        """
        Sets the log_content_type_contains of this SearchClusterLogsDetails.
        The string represent the type of compute logs
        Common values include `events`, `driver`, `executor`.


        :param log_content_type_contains: The log_content_type_contains of this SearchClusterLogsDetails.
        :type: str
        """
        self._log_content_type_contains = log_content_type_contains

    @property
    def subject_contains(self):
        """
        Gets the subject_contains of this SearchClusterLogsDetails.
        The string represent the specific of compute nodes
        Common values include `spark-driver`, `spark-executor-1`.


        :return: The subject_contains of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._subject_contains

    @subject_contains.setter
    def subject_contains(self, subject_contains):
        """
        Sets the subject_contains of this SearchClusterLogsDetails.
        The string represent the specific of compute nodes
        Common values include `spark-driver`, `spark-executor-1`.


        :param subject_contains: The subject_contains of this SearchClusterLogsDetails.
        :type: str
        """
        self._subject_contains = subject_contains

    @property
    def message_contains(self):
        """
        Gets the message_contains of this SearchClusterLogsDetails.
        The string represent a filter to search message string from the log
        Supported only when `logContentTypeContains` is `events`.


        :return: The message_contains of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._message_contains

    @message_contains.setter
    def message_contains(self, message_contains):
        """
        Sets the message_contains of this SearchClusterLogsDetails.
        The string represent a filter to search message string from the log
        Supported only when `logContentTypeContains` is `events`.


        :param message_contains: The message_contains of this SearchClusterLogsDetails.
        :type: str
        """
        self._message_contains = message_contains

    @property
    def log_level(self):
        """
        Gets the log_level of this SearchClusterLogsDetails.
        The severity level of the log messages to filter.
        Common values include `INFO`, `WARN`, `ERROR`, `DEBUG`, etc.
        Supported only when `logContentTypeContains` is `executor` or `driver`.


        :return: The log_level of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._log_level

    @log_level.setter
    def log_level(self, log_level):
        """
        Sets the log_level of this SearchClusterLogsDetails.
        The severity level of the log messages to filter.
        Common values include `INFO`, `WARN`, `ERROR`, `DEBUG`, etc.
        Supported only when `logContentTypeContains` is `executor` or `driver`.


        :param log_level: The log_level of this SearchClusterLogsDetails.
        :type: str
        """
        self._log_level = log_level

    @property
    def opc_request_id(self):
        """
        Gets the opc_request_id of this SearchClusterLogsDetails.
        The unique request identifier used for tracking the API call.
        Supported only when `logContentTypeContains` is `executor` or `driver`.


        :return: The opc_request_id of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._opc_request_id

    @opc_request_id.setter
    def opc_request_id(self, opc_request_id):
        """
        Sets the opc_request_id of this SearchClusterLogsDetails.
        The unique request identifier used for tracking the API call.
        Supported only when `logContentTypeContains` is `executor` or `driver`.


        :param opc_request_id: The opc_request_id of this SearchClusterLogsDetails.
        :type: str
        """
        self._opc_request_id = opc_request_id

    @property
    def advanced_filter(self):
        """
        Gets the advanced_filter of this SearchClusterLogsDetails.
        Contains any key and value that you want to search in the logs.
        Should be provided in the format 'key = value'. As of now supported keys are 'taskRunKey' and 'jobRunKey'


        :return: The advanced_filter of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._advanced_filter

    @advanced_filter.setter
    def advanced_filter(self, advanced_filter):
        """
        Sets the advanced_filter of this SearchClusterLogsDetails.
        Contains any key and value that you want to search in the logs.
        Should be provided in the format 'key = value'. As of now supported keys are 'taskRunKey' and 'jobRunKey'


        :param advanced_filter: The advanced_filter of this SearchClusterLogsDetails.
        :type: str
        """
        self._advanced_filter = advanced_filter

    @property
    def thread(self):
        """
        Gets the thread of this SearchClusterLogsDetails.
        The thread identifier that generated the log.
        Supported only when `logContentTypeContains` is `executor` or `driver`.


        :return: The thread of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._thread

    @thread.setter
    def thread(self, thread):
        """
        Sets the thread of this SearchClusterLogsDetails.
        The thread identifier that generated the log.
        Supported only when `logContentTypeContains` is `executor` or `driver`.


        :param thread: The thread of this SearchClusterLogsDetails.
        :type: str
        """
        self._thread = thread

    @property
    def execution_context_id(self):
        """
        Gets the execution_context_id of this SearchClusterLogsDetails.
        The execution context identifier associated with the log.
        Supported only when `logContentTypeContains` is `executor` or `driver`.


        :return: The execution_context_id of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._execution_context_id

    @execution_context_id.setter
    def execution_context_id(self, execution_context_id):
        """
        Sets the execution_context_id of this SearchClusterLogsDetails.
        The execution context identifier associated with the log.
        Supported only when `logContentTypeContains` is `executor` or `driver`.


        :param execution_context_id: The execution_context_id of this SearchClusterLogsDetails.
        :type: str
        """
        self._execution_context_id = execution_context_id

    @property
    def log_stream_type_contains(self):
        """
        Gets the log_stream_type_contains of this SearchClusterLogsDetails.
        Filters logs based on the output stream type.
        - This includes standard output (`stdout`) for regular program output
        - and standard error (`stderr`) for error messages and diagnostics.
        Only supported when logContentTypeContains=executor/driver.


        :return: The log_stream_type_contains of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._log_stream_type_contains

    @log_stream_type_contains.setter
    def log_stream_type_contains(self, log_stream_type_contains):
        """
        Sets the log_stream_type_contains of this SearchClusterLogsDetails.
        Filters logs based on the output stream type.
        - This includes standard output (`stdout`) for regular program output
        - and standard error (`stderr`) for error messages and diagnostics.
        Only supported when logContentTypeContains=executor/driver.


        :param log_stream_type_contains: The log_stream_type_contains of this SearchClusterLogsDetails.
        :type: str
        """
        self._log_stream_type_contains = log_stream_type_contains

    @property
    def event_type(self):
        """
        Gets the event_type of this SearchClusterLogsDetails.
        Filters logs based on the event type.
        Supported only when `logContentTypeContains` is `events`.


        :return: The event_type of this SearchClusterLogsDetails.
        :rtype: str
        """
        return self._event_type

    @event_type.setter
    def event_type(self, event_type):
        """
        Sets the event_type of this SearchClusterLogsDetails.
        Filters logs based on the event type.
        Supported only when `logContentTypeContains` is `events`.


        :param event_type: The event_type of this SearchClusterLogsDetails.
        :type: str
        """
        self._event_type = event_type

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
